### PR TITLE
Add flake8 to linters

### PR DESCRIPTION
**Pull request by florimondmanca**
This PR is a proposal to add `flake8` to the uvicorn toolchain.

HTTPX and other projects have shown me that having a linter (and not only a formatting fixer) is invualuable for catching subtle bugs and ensuring code quality as a whole. Eg #618 could have been avoided with `flake8` installed.

In any case, happy to pull down if that's not something we want — the original set of check tools added via #290 sticked to black, autoflake and isort, because that's what is in Starlette. (But I'd advocate for adding `flake8` to Starlette, too.)

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -50,7 +48,7 @@ def get_packages(package):
     "uvloop>=0.14.0 ;" + env_marker,
 ]
 
-extras_require = {"watchdogreload": ["watchdog>0.10,<0.11"]},
+extras_require = {"watchdogreload": ["watchdog>0.10,<0.11"]}
```
Comment (florimondmanca) [inline setup.py]: These changes are necessary to make flake8 pass, we should wait for #618.

**Comment by lovelydinosaur** [comment]
Looks great actually yes.

Can you talk me through the bits in `setup.cfg`?
I'd *probably* prefer to avoid another top-level config file if possible, so perhaps we could inline those or something?

**Comment by florimondmanca** [comment]
> Can you talk me through the bits in setup.cfg?
> I'd probably prefer to avoid another top-level config file if possible, so perhaps we could inline those or something?

Indeed, it looks like it should be possible to pass this config as command line arguments to `flake8`.

**Comment by florimondmanca** [comment]
Updated the PR. :-)

One other thing that'd be very nice too would be a `scripts/check`, and making `scripts/install` CI-friendly and platform-accommodating, like we have in HTTPX. We can use them straight in CI, but do the platform switching in the scripts. That way, those scripts work locally on those platforms too… But anyway, that's something else.

**Review by tiangolo — APPROVED** [review]
I love `flake8`! :nerd_face: :heart:

**Comment by florimondmanca** [comment]
Thanks all! 🎉

**Pull request merged**